sweepbatcher/sweep_batcher.go

```diff
@@ -418,7 +418,22 @@ func (b *Batcher) handleSweep(ctx context.Context, sweep *sweep,
 	// can't attach its notifier to the batch as that is no longer running.
 	// Instead we directly detect and return the spend here.
 	if completed && *notifier != (SpendNotifier{}) {
-		return b.monitorSpendAndNotify(ctx, sweep, notifier)
+		// Verify that the parent batch is confirmed. Note that a batch
+		// is only considered confirmed after it has received three
+		// on-chain confirmations to prevent issues caused by reorgs.
+		parentBatch, err := b.store.GetParentBatch(ctx, sweep.swapHash)
```

Comment:
we could pass the `parentBatch` into `monitorSpendAndNotify`, since the method is called in only one place. Then  remove the parent batch call in that method.

Reply:
Good idea, passed in the parent batch ID since that's what we need.